### DISCUSSION TITLE
https://togithub.com/ is currently blocked by OpenDNS

### DISCUSSION THREAD

**Discussion by gr2m** [renovatebot/renovate#27568] [Request Help] [answered]
### How are you running Renovate?

None

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

None

### Describe the problem

<img width="976" alt="image" src="https://github.com/renovatebot/renovate/assets/39992/cd1004f2-9991-4f27-b842-ac6651f6dfdd">

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
Not seeing this from iOS safari browser. Possibly a temporary problem?

**Reply by gr2m** [reply]
I still see it in both chrome and safari on latest Mac OS X, however only on one laptop, not on another 🤷🏼 This is not a blocker for me, feel free to just close the discussion. But if you want me to do something to help debug this, please let me know

**Reply by rarkins** [reply]
`curl` also thinks the cert is ok:

```
❯ curl --verbose https://togithub.com/actions/add-to-project
*   Trying 52.58.254.253:443...
* Connected to togithub.com (52.58.254.253) port 443
* ALPN: curl offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
* (304) (IN), TLS handshake, Unknown (8):
* (304) (IN), TLS handshake, Certificate (11):
* (304) (IN), TLS handshake, CERT verify (15):
* (304) (IN), TLS handshake, Finished (20):
* (304) (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / AEAD-CHACHA20-POLY1305-SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=*.togithub.com
*  start date: Feb  5 20:13:16 2024 GMT
*  expire date: May  5 20:13:15 2024 GMT
*  subjectAltName: host "togithub.com" matched cert's "togithub.com"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://togithub.com/actions/add-to-project
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: togithub.com]
* [HTTP/2] [1] [:path: /actions/add-to-project]
* [HTTP/2] [1] [user-agent: curl/8.4.0]
* [HTTP/2] [1] [accept: */*]
> GET /actions/add-to-project HTTP/2
> Host: togithub.com
> User-Agent: curl/8.4.0
> Accept: */*
> 
< HTTP/2 302 
< age: 0
< cache-status: "Netlify Edge"; fwd=miss
< content-type: text/plain
< date: Tue, 27 Feb 2024 04:56:06 GMT
< location: https://github.com/actions/add-to-project
< netlify-vary: query
< server: Netlify
< strict-transport-security: max-age=31536000
< x-nf-request-id: 01HQMEQ2EYAR9PA5J3KB8DCSXQ
< 
Redirecting to https://github.com/actions/add-to-project
* Connection #0 to host togithub.com left intact
Execution time: 0.23s
```

**Reply by gr2m** [reply]
here is what I get from the machine where it fails

```
curl --verbose https://togithub.com/actions/add-to-project
*   Trying 146.112.61.104:443...
* Connected to togithub.com (146.112.61.104) port 443
* ALPN: curl offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (OUT), TLS alert, unknown CA (560):
* SSL certificate problem: unable to get local issuer certificate
* Closing connection
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (OUT), TLS alert, unknown CA (560):
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

**Reply by viceice** [reply]
the system certificate store is maybe outdated?

**Reply by gr2m** [reply]
I'm sorry I don't know what that means. Is that a problem on my device or a problem with togithub.com? If there is anything I can do to help debugging, please let me know

**Reply by rarkins** [reply]
What type of device/OS is it?

**Reply by gr2m** [reply]
It's a MacBook Pro M1 Max on Mac OS Sonoma 14.3.1

**Reply by Churro** [reply, marked as answer]
@gr2m, are you using OpenDNS by any chance? The host hit-block.opendns.com links to IP address 146.112.61.104 that is shown in your curl output. This hints that you are using OpenDNS with a confugration that blocks access to togithub.com.

**Reply by gr2m** [reply]
I'm getting this now:

<img width="929" alt="image" src="https://github.com/renovatebot/renovate/assets/39992/860b9714-fbbf-4ab2-a619-18f59726560e">

Disabling Umbrella resolves the problem. Here is what I was told by our IT helpdesk:

> Yes I believe we use OpenDNS which is being used by Cisco Umbrella. 
> https://thehub.github.com/security/policy-desk/standards/endpoint-security-standard/#umbrella
>  
> https://umbrella.cisco.com/opendns-cisco-umbrella
> You could try to temporarily disable cisco umbrella to see if the behavior changes when it's on versus off.
>  
> If you need any help with Umbrella you can create an issue in the Umbrella repo. 
> https://github.com/github/umbrella/issues

**Reply by rarkins** [reply]
Is that a private repo (https://github.com/github/umbrella)?

I would be happy to transfer `togithub.com` to GitHub naturally - the community would benefit by being able to link to GitHub endpoints without triggering backlinks

**Reply by gr2m** [reply]
> Is that a private repo (https://github.com/github/umbrella)?

Indeed, sorry I missed that, I'll ask about it myself in the private repo

> I would be happy to transfer `togithub.com` to GitHub naturally - the community would benefit by being able to link to GitHub endpoints without triggering backlinks

Yes I agree, it will be hard to find someone to own it internally. I'll ask about it, might take a while to get feedback on it. I'll ping you when I hear something

**Reply by rarkins** [reply]
I suspect something like https://to.github.com would be better long term, with a 301 permanent redirect from togithub.com to to.github.com. Alternatively having github unblock togithub.com it is suitable :)

**Reply by gr2m** [reply]
fyi: togithub.com was removed from GitHub's Umbrella Block list

**Reply by HonkingGoose** [reply]
I created an issue to track the "transfer ownership of redirect domain to GitHub" part of the discussion:

- https://github.com/renovatebot/renovate/issues/29370
